recipes/mono/mono-osx.recipe

```diff
@@ -0,0 +1,149 @@
+        shutil.copy(os.path.join(self.config.prefix, 'bin', 'mono-sgen'),
+                    os.path.join(self.config.prefix, 'bin', 'mono'))
+        if(os.path.exists(mono_link)):
+            os.remove(mono_link)
+        if(os.path.exists(monosgen_link)):
+            os.remove(monosgen_link)
+        os.symlink(os.path.join(self.config.prefix, 'lib', 'libmonosgen-2.0.1.dylib'),
+                   monosgen_link)
+        os.symlink(monosgen_link,
+                   mono_link)
+
+    def inplace_change(self, filename, old_string, new_string):
+        # Safely read the input filename using 'with'
+        with open(filename) as f:
+            s = f.read()
+            if old_string not in s:
+                print '"{old_string}" not found in {filename}.'.format(**locals())
+                return
+
+        # Safely write the changed content, if found in the file
+        with open(filename, 'w') as f:
+            print 'Changing "{old_string}" to "{new_string}" in {filename}'.format(**locals())
+            s = s.replace(old_string, new_string)
+            f.write(s)
+    
+    def post_install(self):
+        config = os.path.join(self.config.prefix, 'etc', 'mono', 'config')
+        prefix = self.config.libdir
+        if prefix[-1] != '/':
+            prefix += '/'
+        shell.replace(config, {prefix: '',
+            'libintl.dylib': 'libintl.8.dylib',
+            'libintl.so': 'libintl.8.so',
+            'libgdiplus.dylib': 'libgdiplus.0.dylib',
+            'libgdiplus.so': 'libgdiplus.so.0',
+            })
+        shell.call ('rm -rf lib/mono/2.0/nunit*', self.config.prefix)
+        shell.call ('rm -rf lib/mono/4.5/nunit*', self.config.prefix)
+        shell.call ('rm -rf lib/mono/gac/*/2.4.8.0__96d09a1eb7f44a77',
+                self.config.prefix)
+
+    def relocate_osx_libraries(self):
+        super(Recipe, self).relocate_osx_libraries()
+        mono_prefix_dir = '/Library/Frameworks/Mono.framework/Versions/%s' % self.version
+        relocator = OSXRelocator(mono_prefix_dir, mono_prefix_dir, True)
+        relocator.change_id(os.path.join(self.config.prefix, 'lib','libmonosgen-2.0.1.dylib'))
+        relocator.change_id(os.path.join(self.config.prefix, 'lib','libMonoPosixHelper.dylib'))
+        relocator.change_lib_path(os.path.join(self.config.prefix, 'lib','libmonosgen-2.0.1.dylib'),
+            '/usr/lib/libiconv.2.dylib', '@rpath/lib/libiconv.2.dylib')
+        relocator.change_lib_path(os.path.join(self.config.prefix, 'lib','libMonoPosixHelper.dylib'),
+            '/usr/lib/libiconv.2.dylib', '@rpath/lib/libiconv.2.dylib')
+        relocator.change_lib_path(os.path.join(self.config.prefix, 'bin','mono'),
+            '/usr/lib/libiconv.2.dylib', '@rpath/lib/libiconv.2.dylib')
+        relocator.change_lib_path(os.path.join(self.config.prefix, 'bin','mono-sgen'),
+            '/usr/lib/libiconv.2.dylib', '@rpath/lib/libiconv.2.dylib')
```

Comment:
reuse `/usr/lib/libiconv.2.dylib` and `@rpath/lib/libiconv.2.dylib` in variables